user_tools/src/spark_rapids_tools/tools/qualx/model.py

```diff
@@ -196,31 +197,39 @@ def predict(
 
     if 'y' in results_df.columns:
         # reconstruct original gpu duration for validation purposes
-        results_df['gpuDuration'] = results_df['Duration'] / results_df['y']
-        results_df['gpuDuration'] = np.floor(results_df['gpuDuration'])
+        results_df[f'gpu_{label}'] = results_df[label] / results_df['y']
+        results_df[f'gpu_{label}'] = np.floor(results_df[f'gpu_{label}'])
 
     # adjust raw predictions with stage/sqlID filtering of unsupported ops
-    results_df['Duration_pred'] = results_df['Duration'] * (
+    results_df[f'{label}_pred'] = results_df[label] * (
         1.0
         - results_df['fraction_supported']
         + (results_df['fraction_supported'] / results_df['y_pred'])
     )
     # compute fraction of duration in supported ops
-    results_df['Duration_supported'] = (
-        results_df['Duration'] * results_df['fraction_supported']
+    results_df[f'{label}_supported'] = (
+        results_df[label] * results_df['fraction_supported']
     )
     # compute adjusted speedup (vs. raw speedup prediction: 'y_pred')
     # without qual data, this should be the same as the raw 'y_pred'
-    results_df['speedup_pred'] = results_df['Duration'] / results_df['Duration_pred']
+    results_df['speedup_pred'] = results_df[label] / results_df[f'{label}_pred']
     results_df = results_df.drop(columns=['fraction_supported'])
 
     return results_df
 
 
 def extract_model_features(
-    df: pd.DataFrame, split_functions: Mapping[str, Callable[[pd.DataFrame], pd.DataFrame]] = None
+    df: pd.DataFrame,
+    split_functions: Mapping[str, Callable[[pd.DataFrame], pd.DataFrame]] = None,
 ) -> Tuple[pd.DataFrame, List[str], str]:
     """Extract model features from raw features."""
+    label = get_label()
+    expected_model_features = expected_raw_features - ignored_features
+    expected_model_features.remove(label)
+    if label == 'duration_sum':
+        # for 'duration_sum' label, also remove 'duration_mean' since it's related to 'duration_sum'
+        expected_model_features.remove('duration_mean')
```

Comment:
Keeping duration_mean could give opportunity for non-linear speedup estimate based on duration_mean.   Not sure should be removed.

Reply:
duration_mean is [directly computed from duration_sum / numTasks_sum](https://github.com/NVIDIA/spark-rapids-tools/blob/dev/user_tools/src/spark_rapids_tools/tools/qualx/preprocess.py#L537-L543), so I was trying to avoid leaking any (duration_sum) label information in the training features.

That said, the true label would be the ratio of CPU/GPU duration_sum, so it might be ok to leave.

Reply:
Yes the true label has GPU duration.   Anything with CPU info is ok and doesn't leak.   Though may be useless/unnecessary.